xarray/core/types.py

```diff
@@ -182,8 +182,7 @@ def copy(
 DsCompatible = Union["Dataset", "DaCompatible"]
 GroupByCompatible = Union["Dataset", "DataArray"]
 
-Dims = Union[str, Iterable[Hashable], "ellipsis", None]
-OrderedDims = Union[str, Sequence[Union[Hashable, "ellipsis"]], "ellipsis", None]
+Dims = Union[Hashable, Sequence[Hashable], None]
```

Comment:
Am on phone but this has been the subject of much discussion, worth reviewing that...

Reply:
https://github.com/pydata/xarray/issues/6142

So there are good reasons to have `str | Iterable[Hashable]` rather than `Hashable | Iterable[Hashable]`...

Is there a cost of leaving `ellipsis` in? It's nice to have it be explicit, even if it doesn't affect the type resolution?

(+1 to much of the PR though, updating definitions to use `Dims`...)